tests/furusiya/model/components/ai/test_stallion.py

```diff
@@ -0,0 +1,27 @@
+from unittest.mock import MagicMock
+
+import pytest
+
+from model.components.ai.stallion import StallionAi
+
+
+@pytest.fixture
+def stallion():
+    stallion_mock = MagicMock(x=2, y=0)
+    stallion_mock.player = MagicMock(x=4, y=1)
+    stallion_mock.ai = StallionAi(stallion_mock)
+    yield stallion_mock
+
+
+def test_take_turn_far(stallion):
+    stallion.distance_to.return_value = 5
+    stallion.ai.take_turn()
+
+    stallion.move_towards.assert_called_with(stallion.player.x, stallion.player.y)
```

Comment:
Not enough. This is white-box testing (testing what a method calls internally). Please check that the stallion actually moved closer to the player.

Reply:
Thing is, we're using a mock here. What's called isn't the original GameObject `move_towards`.
I guess I can make a lambda emulating what it would do.